pythonosc/osc_message.py

```diff
@@ -33,8 +33,11 @@ def _parse_datagram(self):
       if type_tag.startswith(','):
         type_tag = type_tag[1:]
 
+      params = []
+      param_stack = [params]
       # Parse each parameter given its type.
       for param in type_tag:
+        have_val = True
```

Comment:
I'd prefer an array specific approach here instead of this generic one:
is_within_array = False
or something like that